src/main/java/com/google/sps/data/Instance.java

```diff
@@ -0,0 +1,113 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+import java.util.Optional;
+import com.google.sps.data.EntityConstants;
+
+/** Class that represents an instance. */
+public class Instance implements InstanceInterface {
+  
+  // The name of this instance.
+  private final String instanceName;
+
+  // The state this instance is in. 
+  // https://cloud.google.com/compute/docs/instances/instance-life-cycle
+  private String state;
+
+  // The session associated with this instance. If sessionId is not present,
+  // then the VM is not being used for a session.
+  private Optional<String> sessionId;
+
+  /** Initializes an Instance object
+   * @param {String} instanceName - the name of the instance.
+   * @param {String} state - the state the instance is in.
+   * @param {Optional<String>} sessionId - the session id 
+   *    associated with the instance. 
+   */
+  public Instance(String instanceName, String state, 
+  Optional<String> sessionId) {
+    this.instanceName = instanceName;
+    this.state = state;
+    this.sessionId = sessionId;
+  }
+
+  public String getInstanceName() {
+    return instanceName;
+  } 
+
+  public String getState() {
+    return state;
+  }
+
+  public Optional<String> getSessionId() {
+    return sessionId;
+  }
+
+  public void setState(String state) {
+    this.state = state;
+  }
+
+  public void setSessionId(Optional<String> sessionId) {
```

Comment:
This shouldnt be Optional. If someone is setting a sessionId, its not optional lol they are expected to be giving us a real thing.

Reply:
It gives type error if it is not passed as Optional since that's how sessionId is defined.

Reply:
Unless setSessionId converts it to Optional then assign it to this.sessionId